homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,304 @@
+"""Support for Overkiz alarm control panel."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+from pyoverkiz.enums.ui import UIWidget
+from pyoverkiz.types import StateType as OverkizStateType
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+)
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_ARM_NIGHT,
+    SUPPORT_ALARM_TRIGGER,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN
+from .entity import OverkizDescriptiveEntity
+
+
+@dataclass
+class OverkizAlarmDescriptionMixin:
+    """Define an entity description mixin for switch entities."""
+
+    supported_features: int
+    fn_state: Callable[[Callable[[str], OverkizStateType]], str]
+
+
+@dataclass
+class OverkizAlarmDescription(
+    AlarmControlPanelEntityDescription, OverkizAlarmDescriptionMixin
+):
+    """Class to describe an Overkiz alarm control panel."""
+
+    alarm_disarm: Callable[
+        [Callable[..., Awaitable[None]]], Awaitable[None]
+    ] | None = None
+    alarm_arm_home: Callable[
+        [Callable[..., Awaitable[None]]], Awaitable[None]
+    ] | None = None
+    alarm_arm_night: Callable[
+        [Callable[..., Awaitable[None]]], Awaitable[None]
+    ] | None = None
+    alarm_arm_away: Callable[
+        [Callable[..., Awaitable[None]]], Awaitable[None]
+    ] | None = None
+    alarm_trigger: Callable[
+        [Callable[..., Awaitable[None]]], Awaitable[None]
+    ] | None = None
```

Comment:
I feel something like this would be cleaner:
```suggestion
    alarm_disarm: str | None = None
    alarm_disarm_args: Any = UNDEFINED
    alarm_arm_home: str | None = None
    alarm_arm_home_args: Any = UNDEFINED
    alarm_arm_night: str | None = None
    alarm_arm_night_args: Any = UNDEFINED
    alarm_arm_away: str | None = None
    alarm_arm_away_args: Any = UNDEFINED
    alarm_trigger: str | None = None
    alarm_trigger_args: Any = UNDEFINED
```